### PR TITLE
Use upstream RPC-MaaS (mitaka)

### PR DESCRIPTION
**Pull request by cloudnull**
This change converts our deployment of RPC MaaS from the old role method
to the new playbook method. This change will provide us access to a
faster and more flexible monitoring solution and allow all customers to
benifit from Monitoring as a Service using a unified code base.

The maas group_vars have been removed and the values of the old file
have been placed into the `user_osa_variables_defaults.yml` file. This
has been done to allow the rpc-maas playbooks to inherit from the user
variables in cases where the group vars not not available.

Signed-off-by: Kevin Carter <kevin.carter@rackspace.com>
(cherry picked from commit dbd0253e619710b926230e573c9f273559102bf1)

### CODE REVIEW

**Comment by odyssey4me** [comment]
recheck_cit_all

**Comment by odyssey4me** [comment]
https://github.com/rcbops/rpc-openstack/pull/2349 will need to merge to get travis passing

**Comment by odyssey4me** [comment]
This patch needs to be rebased in order to get travis passing.

**Comment by mattt416** [comment]
The deploy jobs failed with `ERROR: file could not read: /opt/rpc-maas/playbooks/site.yml`, is rpc-maas not getting sucked down correctly?

**Comment by cloudnull** [comment]
@mattt416 yes, the maas-get playbook was not being run, I've updated the patches to fix this.

File: scripts/deploy.sh [hunk 1]
```diff
@@ -177,11 +177,16 @@ openstack-ansible -i "localhost," patcher.yml
 
 # set permissions and lay down overrides files
 chmod 0440 /etc/openstack_deploy/user_*_defaults.yml
-if [[ ! -f /etc/openstack_deploy/user_osa_variables_overrides.yml ]]; then
-  cp "${RPCD_DIR}"/etc/openstack_deploy/user_osa_variables_overrides.yml /etc/openstack_deploy/user_osa_variables_overrides.yml
+if [[ ! -f "${OA_OVERRIDES}" ]]; then
+  cp "${RPCD_DIR}/etc/openstack_deploy/user_osa_variables_overrides.yml" "${OA_OVERRIDES}"
```
Comment (evrardjp) [inline scripts/deploy.sh]: Do we need this?

File: rpcd/etc/openstack_deploy/user_rpco_variables_overrides.yml [hunk 2]
```diff
@@ -20,3 +20,10 @@
 
 # this variable is here because vars files need to have at least ONE variable defined.
```
Comment (evrardjp) [inline rpcd/etc/openstack_deploy/user_rpco_variables_overrides.yml]: Is that the proper file?

File: scripts/deploy.sh (hunk 1 above)
Comment (cloudnull) [inline scripts/deploy.sh]: reverted.

File: rpcd/etc/openstack_deploy/user_rpco_variables_overrides.yml (hunk 2 above)
Comment (cloudnull) [inline rpcd/etc/openstack_deploy/user_rpco_variables_overrides.yml]: yes.

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all

**Comment by cloudnull** [comment]
recheck_cit_all -- tempest is failing

**Comment by cloudnull** [comment]
recheck_cit_trusty_majorupgrade_ceph
recheck_cit_trusty_majorupgrade_swift

**Comment by cloudnull** [comment]
recheck_cit_trusty_majorupgrade_ceph
recheck_cit_trusty_majorupgrade_swift

**Comment by cloudnull** [comment]
recheck_cit_trusty_majorupgrade_ceph
recheck_cit_trusty_majorupgrade_swift

**Comment by alextricity25** [comment]
recheck_cit_trusty_majorupgrade_ceph
recheck_cit_trusty_majorupgrade_swift

**Pull request merged**